_data-prepper/pipelines/configuration/processors/mutate-event.md

```diff
@@ -0,0 +1,302 @@
+
+```json
+{"message2": "value2"}
+```
+
+> If `message` does not exist in the event, then no action occurs.
+
+
+## RenameKey
+
+The `rename key` processor renames keys in an event.
+
+### Configuration
+
+You can configure the `rename key` processor with the following options.
+
+Option | Required | Description
+:--- | :--- | :---
+| `entries` | Yes | A list of entries to rename in an event. | 
+| `from_key` | Yes | The key of the entry to be renamed. |
+| `to_key` | Yes | The new key of the entry.
+| `overwrite_if_to_key_exists` | No | When set to a value of`true`, if `to_key` already exists in the event, then the existing value will be overwritten. The default value is `false`. |
+
+### Usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - rename_keys:
+        entries:
+        - from_key: "message"
+          to_key: "newMessage"
+          overwrite_if_to_key_exists: true
+  sink:
+    - stdout:
+```
+
+Next, create a log file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with this filepath. For more information, see [Configuring Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/getting-started/#2-configuring-data-prepper).
+
+```json
+{"message": "value"}
+```
+
+When you run the `RenameKey` processor, it parses the message into the following output as the "newMessage":
+
+```json
+{"newMessage": "value"}
+```
+
+> If `newMessage` already exists, its existing value is overwritten with `value`.
+
+
+
+### Special considerations
+
+Renaming operations occur in the order that the key pair entries are listed in within the `pipelines.yaml` file. This means that chaining (where key pairs are renamed in sequence) is implicit with the `RenameKey` processor. See the following `piplines.yaml` file example:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - rename_key:
+        entries:
+        - from_key: "message"
+          to_key: "message2"
+        - from_key: "message2"
+          to_key: "message3"
+  sink:
+    - stdout:
+```
+
+Add the following contents to the `logs_json.log` file:
+
+```json
+{"message": "value"}
+```
+
+After the processor runs, the following output appears:
+
+```json
+{"message3": "value"}
+```
+
+## ConvertEntry
+
+The `ConvertEntry` processor converts a value type associated with the specified key in a message to the specified type. It is a casting processor that changes the types of some fields in the event or message. Some of inputted data may need to be converted to different types, such as an integer or a double, or a string to an integer, so that it will pass the events through condition-based processors, or to perform conditional routing.
```

Comment:
"Some entered data may need to be ..."

"...converted to different types, such as an integer **to** a double, or a string to an integer" Correct change?

"... so that it will pass the events through condition-based processors or perform conditional routing."